### DISCUSSION TITLE
Has anyone tried setting SQS to trigger for c7n-mailer lambda instead of polling SQS

### DISCUSSION THREAD

**Discussion by nitrocode** [cloud-custodian/cloud-custodian#9686] [AWS]
https://aws.amazon.com/blogs/aws/aws-lambda-adds-amazon-simple-queue-service-to-supported-event-sources/

Currently my `lambda_schedule` is set to poll the SQS every 5 minutes which means 12 times per hour, 288 times a month.

We don't get a lot of alerts (maybe one a day) and when we do, we'd rather get them immediately. Has anyone tried this? If so, did it work well? And is there a way to disable the schedule?

**Comment by ajkerrigan** [comment]
Not yet that I know of. It has come up in discussion before (tracked in https://github.com/cloud-custodian/cloud-custodian/issues/2579), but from what I've seen it has remained in the realm of "this sounds neat and not too hard to implement, but isn't a priority".

**Reply by nitrocode** [reply]
It seems like it can be done on the sqs side and I believe the SQS is a prerequisite for setting up c7n-mailer.

https://docs.aws.amazon.com/AWSSimpleQueueService/latest/SQSDeveloperGuide/sqs-configure-lambda-function-trigger.html

So the only change we'd need in c7n-mailer is a way to disable the schedule or set it to a difficult to hit cron.

E.g. every 20 years
```
0 0 0 1 1 ? 2024/20
```

**Reply by ajkerrigan** [reply]
Yeah, I would expect that the SQS trigger would be an opt-in replacement for the existing time-based trigger. And in those direct trigger cases, we'd need to suck messages out of the triggering event rather than explicitly polling the queue.

Might also need to expose a setting for configuring concurrency to avoid having things go off the rails in large bursty environments, but that doesn't seem strictly necessary for a first pass.
